native/java/org/jpype/JPypeUtilities.java

```diff
@@ -0,0 +1,21 @@
+package org.jpype;
+
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class JPypeUtilities
+{
+  public static Path getJarPath(Class c)
+  {
+    try
+    {
+      return Paths.get(c.getProtectionDomain().getCodeSource().getLocation()
```

Comment:
none of this call may return null?

Reply:
This is code for the Python from Java start (not currently).  It is being called once to find the location of the org.jpype.jar so we can find the _jpype.<arch>.so file.   If it fails then the whole system fails to load which is the intended behavior.    getCodeSource() can fail according to the docs, but it will be a fatal error either way.